www/src/Services/Http.ts

```diff
@@ -1,21 +1,30 @@
 // This is modeled after Axios to make an easier transition, so this may need more buildout
 // if more than simple GET and POST requests are required.
 
+type GetOptions = {
+	headers?: Record<string, string>,
+	signal?: AbortSignal
+}
+
 class Http {
-	async get(url: string, headers: object = {}) {
+	async get(url: string, { headers = {}, signal }: GetOptions = {}) {
 		try {
 			const response = await fetch(url, {
 				method: 'GET',
 				headers: {
 					'Content-Type': 'application/json',
 					...headers,
 				},
+				signal
 			});
 
 			const json = await response.json();
 			return Promise.resolve({ data: json });
 		}
 		catch (err) {
+			if (signal?.aborted) {
+				throw new Error('ERR_CANCELED');
```

Comment:
I don't think this is correct, fetch should already throw if aborted.

https://dom.spec.whatwg.org/#interface-abortcontroller

Reply:
I don't think is a matter of correctness, yes is throwing and indeed we are in the catch block, the special case here was done to stick to 'ERR_CANCELED' that is a sort of custom identifier to identify the user abort operation. There are alternatives to do so, this was the less risky.

As i mentioned above the error returned changes between firefox and chrome. They are bot DOMException but different message

<img width="1110" alt="image" src="https://github.com/OpenStickCommunity/GP2040-CE/assets/1194048/ecfcf4ca-9274-4709-a085-1a1c06ea6405">

So if we don't like throwing our own error we can either:
- 1 use `err instanceof DOMException` as an identifier
- 2 use err.message containing the word 'abort' generic, hoping that is common to every browser as an identifier
- 3 do not use any identifier and call `abortGetHeldPins` after any error of `getHeldPins` but i m not sure of the implications
- 4 instead of calling 'abort()' we could call abort('-our-custom-abort-message') and that would make the fetch call reject a string instead of an error and we can just match the string ( in that case please pick a string ).


As a developer the current option and the 4th alternative option seems the best to me, the DOMException option is fine too with some risk of overlap from other errors that may happen.

Definetely your choice

Reply:
Oh i see now your edit of the name, I didn't think of using the instance class name. that is probably the simples

Reply:
When identifying DOMException i thought one used the property "name", is there a discrepancy between browsers there also?